# GeoSlide Project — Phase 1: Dataset Audit

**Dataset:** Wireless Sensor Network (WSN) Landslide Dataset
**Notebook:** `01_dataset_audit.ipynb`
**Prepared by:** Software Engineer, GeoSlide Project
**Objective:** Perform a structured, non-destructive audit of the raw dataset prior to any preprocessing, feature engineering, or modeling work.

---

### Scope of this Notebook

This notebook is strictly limited to **Phase 1 — Dataset Audit**. It covers:

- Loading and inspecting the raw dataset
- Reviewing structural information (columns, data types, shape)
- Quantifying missing values and duplicate records
- Reviewing the target variable distribution
- Summarizing descriptive statistics of numerical features
- Producing a final audit summary

**Out of scope (explicitly not performed here):** data cleaning, imputation, encoding, scaling, feature engineering, exploratory data analysis beyond the audit, and model training. These will be addressed in later phases of the GeoSlide pipeline.

---

## 1. Import Libraries

We import the core libraries required for data loading, numerical operations, and visualization. Warnings are suppressed to keep the notebook output clean and readable for reporting purposes.

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import plotly.express as px
import warnings

warnings.filterwarnings("ignore")

pd.set_option("display.max_columns", None)
pd.set_option("display.width", 120)

print("Libraries imported successfully.")

Libraries imported successfully.


## 2. Load Dataset

The raw CSV file is loaded directly into a Pandas DataFrame without any modification. We then inspect the shape of the dataset along with the first and last few records to get an initial sense of its structure.

> **Note (Google Colab):** If running in Colab, upload `wsn_landslide_data.csv` to the Colab file system (e.g. via `files.upload()` or by mounting Google Drive) and update `DATA_PATH` accordingly.

In [2]:
# Path to the raw dataset (update this path if running in Google Colab)
DATA_PATH = "wsn_landslide_data.csv"

df = pd.read_csv(DATA_PATH)

print("Dataset loaded successfully.")

Dataset loaded successfully.


In [3]:
# Shape of the dataset
print(f"Dataset shape: {df.shape[0]} rows x {df.shape[1]} columns")

Dataset shape: 9864 rows x 35 columns


In [4]:
# First 5 records
df.head()

,Rainfall_mm,Slope_Angle,Soil_Saturation,Vegetation_Cover,Rainfall_3Day,Rainfall_7Day,Aspect,Elevation_m,NDVI_Index,Land_Use_Urban,Land_Use_Forest,Land_Use_Agriculture,Earthquake_Activity,Proximity_to_Water,Distance_to_Road_m,Temperature_C,Humidity_percent,Soil_pH,Clay_Content,Sand_Content,Silt_Content,Soil_Erosion_Rate,Historical_Landslide_Count,Soil_Type_Gravel,Soil_Type_Sand,Soil_Type_Silt,Soil_Type_Clay,Pore_Water_Pressure_kPa,Soil_Moisture_Content,Microseismic_Activity,Acoustic_Emission_dB,Soil_Strain,Soil_Temperature_C,TDR_Reflection_Index,Label
0,164.695364,59.783332,0.821479,0.107339,260.138381,79.297169,346.674199,733.776448,0.191948,1,1,1,5.390869,0.847097,961.823634,12.887887,26.236892,7.102161,39.934922,57.743254,24.562387,14.792175,0,1,1,1,0,133.943194,0.143732,0.290945,51.021834,0.005167,22.760036,0.799847,1
1,34.908086,15.153084,0.100428,0.960150,510.295547,247.923576,104.462371,467.708643,0.798321,1,0,0,3.554055,0.077767,138.904464,12.864021,64.196194,7.912667,48.780193,30.285331,36.973295,49.102736,1,0,1,0,1,90.788608,0.266484,0.651758,39.837282,0.003443,15.558373,1.181071,0
2,38.761727,13.135384,0.286526,0.833608,297.730266,194.327012,336.671287,1880.826807,0.479456,1,0,1,4.532214,0.157258,792.351115,6.523798,34.223394,6.463165,55.134475,78.713096,35.814410,47.171786,1,1,0,0,1,83.041150,0.129426,0.440714,68.902366,0.009999,6.205760,1.184971,0
3,32.199977,10.674734,0.255230,0.847569,231.640610,295.139546,300.742864,964.080336,-0.084314,1,1,0,3.069054,0.203603,455.791293,16.244057,38.654212,6.822879,54.431267,52.885510,35.186064,29.573396,1,1,1,1,0,196.089305,0.240198,0.794001,80.196960,0.003850,25.486545,0.677944,0
4,218.114032,48.839269,0.720071,0.018383,330.278249,301.288824,155.550502,165.699102,0.810869,0,0,0,5.623601,0.984898,317.396619,13.847538,95.050941,8.246586,21.311863,46.406911,16.893544,42.128333,1,1,0,1,1,106.778890,0.345724,0.009160,99.919786,0.003061,7.270319,0.882642,1


In [5]:
# Last 5 records
df.tail()

,Rainfall_mm,Slope_Angle,Soil_Saturation,Vegetation_Cover,Rainfall_3Day,Rainfall_7Day,Aspect,Elevation_m,NDVI_Index,Land_Use_Urban,Land_Use_Forest,Land_Use_Agriculture,Earthquake_Activity,Proximity_to_Water,Distance_to_Road_m,Temperature_C,Humidity_percent,Soil_pH,Clay_Content,Sand_Content,Silt_Content,Soil_Erosion_Rate,Historical_Landslide_Count,Soil_Type_Gravel,Soil_Type_Sand,Soil_Type_Silt,Soil_Type_Clay,Pore_Water_Pressure_kPa,Soil_Moisture_Content,Microseismic_Activity,Acoustic_Emission_dB,Soil_Strain,Soil_Temperature_C,TDR_Reflection_Index,Label
9859,186.911121,56.873152,0.885837,0.058470,375.105005,425.433314,174.140151,2636.990436,0.029085,0,1,0,3.426323,0.981562,976.459423,23.211815,52.006322,7.273136,17.943885,73.674736,59.271372,18.383579,1,0,0,0,1,127.861554,0.453029,0.533627,42.456657,0.002680,29.340929,0.615155,1
9860,268.464823,58.468917,0.817995,0.002593,128.357277,720.915314,340.894433,2452.423236,0.710835,1,1,1,4.327940,0.330978,663.116003,26.465554,22.686589,7.797910,17.711012,53.338735,51.572871,6.474193,0,0,1,1,1,114.605703,0.430333,0.465290,75.320910,0.006318,5.761028,1.206497,1
9861,179.059318,60.158346,0.809452,0.214622,195.876243,700.038421,352.343573,1259.108332,0.155847,1,1,1,2.966766,0.015003,155.269239,22.847882,92.077723,6.541440,31.233736,29.257906,18.342645,37.122225,0,0,1,0,0,74.176648,0.477707,0.976331,33.015766,0.005100,15.736253,1.278428,1
9862,38.776381,19.481730,0.238752,0.758117,363.021672,622.604053,76.971637,267.793157,0.844559,0,1,1,2.830920,0.762563,161.760899,34.781780,94.477035,5.893382,45.234293,75.150600,8.838202,38.739480,1,0,1,1,1,49.250911,0.302732,0.107931,32.801850,0.003456,19.502418,0.826547,0
9863,268.351627,50.082374,0.726556,0.259514,317.409414,12.399728,143.659736,2973.346129,0.102796,1,1,1,2.573211,0.803990,928.035395,7.468036,48.255624,6.503789,20.592017,24.427100,27.763778,27.777355,1,1,1,1,1,29.597905,0.318636,0.632004,49.949531,0.000889,17.565231,0.825514,1


## 3. Dataset Information

This section reviews the structural characteristics of the dataset: column names, data types, non-null counts, and summary statistics for all columns. This provides a baseline understanding of the dataset's composition before any further processing is considered.

In [6]:
# Structural information: dtypes, non-null counts, memory usage
df.info()

<class 'pandas.DataFrame'>
RangeIndex: 9864 entries, 0 to 9863
Data columns (total 35 columns):
 #   Column                      Non-Null Count  Dtype  
---  ------                      --------------  -----  
 0   Rainfall_mm                 9864 non-null   float64
 1   Slope_Angle                 9864 non-null   float64
 2   Soil_Saturation             9864 non-null   float64
 3   Vegetation_Cover            9864 non-null   float64
 4   Rainfall_3Day               9864 non-null   float64
 5   Rainfall_7Day               9864 non-null   float64
 6   Aspect                      9864 non-null   float64
 7   Elevation_m                 9864 non-null   float64
 8   NDVI_Index                  9864 non-null   float64
 9   Land_Use_Urban              9864 non-null   int64  
 10  Land_Use_Forest             9864 non-null   int64  
 11  Land_Use_Agriculture        9864 non-null   int64  
 12  Earthquake_Activity         9864 non-null   float64
 13  Proximity_to_Water          9864 non-null   

In [7]:
# Statistical summary of all columns
df.describe(include="all").T

,count,mean,std,min,25%,50%,75%,max
Rainfall_mm,9864.0,125.643618,105.885846,0.000582,24.921180,99.996551,227.181479,299.985147
Slope_Angle,9864.0,36.023345,25.150113,5.000792,12.311669,29.998160,59.678277,79.987824
Soil_Saturation,9864.0,0.524998,0.333087,0.100032,0.199151,0.499968,0.850183,0.999937
Vegetation_Cover,9864.0,0.476049,0.341999,0.000054,0.150170,0.449941,0.804152,0.999970
Rainfall_3Day,9864.0,298.324654,174.136992,0.010042,145.876834,295.410901,452.111577,599.983288
Rainfall_7Day,9864.0,503.397364,287.807884,0.008432,257.167899,508.771270,751.552823,999.939696
Aspect,9864.0,179.656854,103.635389,0.097244,90.790915,179.345879,269.961378,359.888356
Elevation_m,9864.0,1554.277930,830.193051,100.377033,833.122929,1560.377189,2272.900742,2999.637788
NDVI_Index,9864.0,0.401557,0.290701,-0.099820,0.147099,0.404669,0.658994,0.899858
Land_Use_Urban,9864.0,0.503751,0.500011,0.000000,0.000000,1.000000,1.000000,1.000000


In [8]:
# Column names
print("Column Names:")
print(df.columns.tolist())

Column Names:
['Rainfall_mm', 'Slope_Angle', 'Soil_Saturation', 'Vegetation_Cover', 'Rainfall_3Day', 'Rainfall_7Day', 'Aspect', 'Elevation_m', 'NDVI_Index', 'Land_Use_Urban', 'Land_Use_Forest', 'Land_Use_Agriculture', 'Earthquake_Activity', 'Proximity_to_Water', 'Distance_to_Road_m', 'Temperature_C', 'Humidity_percent', 'Soil_pH', 'Clay_Content', 'Sand_Content', 'Silt_Content', 'Soil_Erosion_Rate', 'Historical_Landslide_Count', 'Soil_Type_Gravel', 'Soil_Type_Sand', 'Soil_Type_Silt', 'Soil_Type_Clay', 'Pore_Water_Pressure_kPa', 'Soil_Moisture_Content', 'Microseismic_Activity', 'Acoustic_Emission_dB', 'Soil_Strain', 'Soil_Temperature_C', 'TDR_Reflection_Index', 'Label']


In [9]:
# Data types per column
dtype_summary = df.dtypes.value_counts().reset_index()
dtype_summary.columns = ["Data Type", "Column Count"]
dtype_summary

,Data Type,Column Count
0,float64,26
1,int64,9


## 4. Missing Values

We quantify missing values across all columns, both as raw counts and as a percentage of total records. A visualization is included to make gaps in the data immediately apparent (or to confirm their absence).

In [10]:
# Missing value counts and percentages per column
missing_count = df.isnull().sum()
missing_percent = (df.isnull().sum() / len(df)) * 100

missing_df = pd.DataFrame({
    "Missing Count": missing_count,
    "Missing Percentage (%)": missing_percent.round(2)
}).sort_values(by="Missing Count", ascending=False)

missing_df

,Missing Count,Missing Percentage (%)
Rainfall_mm,0,0.0
Slope_Angle,0,0.0
Soil_Saturation,0,0.0
Vegetation_Cover,0,0.0
Rainfall_3Day,0,0.0
Rainfall_7Day,0,0.0
Aspect,0,0.0
Elevation_m,0,0.0
NDVI_Index,0,0.0
Land_Use_Urban,0,0.0


In [11]:
# Total missing values in the dataset
total_missing = df.isnull().sum().sum()
print(f"Total missing values in dataset: {total_missing}")
print(f"Overall missing percentage: {(total_missing / (df.shape[0] * df.shape[1])) * 100:.4f}%")

Total missing values in dataset: 0
Overall missing percentage: 0.0000%


In [12]:
# Visualize missing values per column
fig = px.bar(
    missing_df.reset_index().rename(columns={"index": "Column"}),
    x="Column",
    y="Missing Percentage (%)",
    title="Missing Values by Column (%)",
    text="Missing Count",
    color="Missing Percentage (%)",
    color_continuous_scale="Reds"
)
fig.update_layout(xaxis_tickangle=-60, template="plotly_white", height=500)
fig.show()

## 5. Duplicate Records

We check for fully duplicated rows in the dataset. Duplicate sensor readings could indicate logging errors or repeated transmissions and are relevant to note during the audit, even though no rows are removed at this stage.

In [13]:
# Check for duplicate records
duplicate_count = df.duplicated().sum()
duplicate_percent = (duplicate_count / len(df)) * 100

print(f"Number of duplicate records: {duplicate_count}")
print(f"Percentage of duplicate records: {duplicate_percent:.2f}%")

Number of duplicate records: 0
Percentage of duplicate records: 0.00%


## 6. Target Variable

The target column for the GeoSlide landslide prediction task is `Label`, where each record is classified into one of the landslide risk/event classes recorded by the wireless sensor network. We examine class counts and proportions to assess class balance.

In [14]:
# Identify the target/label column
TARGET_COLUMN = "Label"

print(f"Target column: '{TARGET_COLUMN}'")

Target column: 'Label'


In [15]:
# Class counts
class_counts = df[TARGET_COLUMN].value_counts().sort_index()
class_percent = (df[TARGET_COLUMN].value_counts(normalize=True).sort_index() * 100).round(2)

target_summary = pd.DataFrame({
    "Class": class_counts.index,
    "Count": class_counts.values,
    "Percentage (%)": class_percent.values
})

target_summary

,Class,Count,Percentage (%)
0,0,4905,49.73
1,1,4959,50.27


In [16]:
# Bar chart of target class distribution
fig = px.bar(
    target_summary,
    x="Class",
    y="Count",
    text="Count",
    color="Class",
    title="Target Variable (Label) Class Distribution",
    labels={"Class": "Label", "Count": "Number of Records"}
)
fig.update_traces(textposition="outside")
fig.update_layout(template="plotly_white", height=450, xaxis=dict(type="category"))
fig.show()

## 7. Numerical Features

This section provides descriptive statistics for all numerical features in the dataset (sensor readings such as rainfall, soil saturation, pore water pressure, seismic activity, etc.). These statistics describe the raw, unmodified distribution of each feature.

In [17]:
# Identify numerical columns
numerical_cols = df.select_dtypes(include=[np.number]).columns.tolist()
print(f"Number of numerical columns: {len(numerical_cols)}")
print(numerical_cols)

Number of numerical columns: 35
['Rainfall_mm', 'Slope_Angle', 'Soil_Saturation', 'Vegetation_Cover', 'Rainfall_3Day', 'Rainfall_7Day', 'Aspect', 'Elevation_m', 'NDVI_Index', 'Land_Use_Urban', 'Land_Use_Forest', 'Land_Use_Agriculture', 'Earthquake_Activity', 'Proximity_to_Water', 'Distance_to_Road_m', 'Temperature_C', 'Humidity_percent', 'Soil_pH', 'Clay_Content', 'Sand_Content', 'Silt_Content', 'Soil_Erosion_Rate', 'Historical_Landslide_Count', 'Soil_Type_Gravel', 'Soil_Type_Sand', 'Soil_Type_Silt', 'Soil_Type_Clay', 'Pore_Water_Pressure_kPa', 'Soil_Moisture_Content', 'Microseismic_Activity', 'Acoustic_Emission_dB', 'Soil_Strain', 'Soil_Temperature_C', 'TDR_Reflection_Index', 'Label']


In [18]:
# Descriptive statistics for numerical features
numerical_stats = df[numerical_cols].describe().T
numerical_stats["range"] = numerical_stats["max"] - numerical_stats["min"]
numerical_stats.round(3)

,count,mean,std,min,25%,50%,75%,max,range
Rainfall_mm,9864.0,125.644,105.886,0.001,24.921,99.997,227.181,299.985,299.985
Slope_Angle,9864.0,36.023,25.150,5.001,12.312,29.998,59.678,79.988,74.987
Soil_Saturation,9864.0,0.525,0.333,0.100,0.199,0.500,0.850,1.000,0.900
Vegetation_Cover,9864.0,0.476,0.342,0.000,0.150,0.450,0.804,1.000,1.000
Rainfall_3Day,9864.0,298.325,174.137,0.010,145.877,295.411,452.112,599.983,599.973
Rainfall_7Day,9864.0,503.397,287.808,0.008,257.168,508.771,751.553,999.940,999.931
Aspect,9864.0,179.657,103.635,0.097,90.791,179.346,269.961,359.888,359.791
Elevation_m,9864.0,1554.278,830.193,100.377,833.123,1560.377,2272.901,2999.638,2899.261
NDVI_Index,9864.0,0.402,0.291,-0.100,0.147,0.405,0.659,0.900,1.000
Land_Use_Urban,9864.0,0.504,0.500,0.000,0.000,1.000,1.000,1.000,1.000


## 8. Dataset Summary

### Audit Overview

| Metric | Value |
|---|---|
| **Number of Rows** | 9864 |
| **Number of Columns** | 35 |
| **Target Column** | `Label` |
| **Target Classes** | 2 |
| **Total Missing Values** | 0 |
| **Duplicate Records** | 0 |

### Target Class Distribution

| Class | Count | Percentage |
|---|---|---|
| 0 | 4905 | 49.73% |
| 1 | 4959 | 50.27% |

### Overall Observations

- The dataset contains **9864 records** across **35 columns**, comprising sensor-derived environmental, geotechnical, and seismic features collected via a wireless sensor network.
- The target variable `Label` is used to represent the landslide classification outcome and contains **2 distinct class(es)**.
- The dataset shows **0 missing value(s)** in total, indicating a complete, well-logged sensor dataset with no gaps to address.
- **0 duplicate record(s)** were identified in the raw data, indicating no redundant sensor transmissions in the raw log.
- The class distribution between labels is reasonably balanced, which is a relevant consideration for the modeling phase (to be addressed later, outside the scope of this notebook).
- Numerical features span a wide range of physical units (e.g. millimeters, degrees Celsius, kPa, dB), suggesting that feature scaling will likely be necessary in a later preprocessing phase.
- No data was cleaned, transformed, or removed as part of this audit — the dataset remains in its original, raw state.

---

**Phase 1 (Dataset Audit) complete.** Findings from this audit should inform the design of the subsequent preprocessing and feature engineering phases of the GeoSlide pipeline.